src/app/static/src/app/components/modelRun/ModelRun.vue

```diff
@@ -9,32 +9,51 @@
         </h4>
         <modal :open="running">
             <h4>Running model</h4>
+            <b-progress :value="progress" :max="1" show-progress animated></b-progress>
         </modal>
     </div>
 </template>
 
 <script lang="ts">
     import Vue from "vue";
-    import {mapActions, mapState} from "vuex";
     import {ModelSubmitParameters} from "../../generated";
-    import {ModelRunState, ModelRunStatus} from "../../store/modelRun/modelRun";
+    import {ModelRunState} from "../../store/modelRun/modelRun";
     import Modal from "../Modal.vue";
     import Tick from "../Tick.vue";
+    import {mapActionsByNames, mapGettersByNames, mapStateProps} from "../../utils";
+    import {BProgress} from "bootstrap-vue";
+
+    interface ComputedState {
+        runId: string
+        success: boolean
+    }
+
+    interface ComputedGetters {
+        running: boolean
+    }
+
+    interface Computed extends ComputedGetters, ComputedState {}
+
+    interface Methods {
+        run: (params: ModelSubmitParameters) => void;
+        poll: (runId: string) => void;
+        runModelWithParams: () => void;
+    }
 
     const namespace: string = 'modelRun';
 
-    export default Vue.extend<any, any, any, any>({
+    export default Vue.extend<{}, Methods, Computed, {}>({
         name: "ModelRun",
-        computed: mapState<ModelRunState>(namespace, {
-            runId: state => state.modelRunId,
-            running: state => state.status == ModelRunStatus.Started,
-            success: state => state.success
-        }),
-        methods: {
-            ...mapActions({
-                run: 'modelRun/run',
-                poll: 'modelRun/poll'
+        computed: {
+            ...mapStateProps<ModelRunState, keyof ComputedState>(namespace, {
+                runId: state => state.modelRunId,
+                success: state => state.status.success,
+                progress: state => parseFloat(state.status.progress)
```

Comment:
Progress is being returned as a string at the moment. I think once https://vimc.myjetbrains.com/youtrack/issue/mrc-578 is done it will become a decimal.